clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java

```diff
@@ -459,6 +459,17 @@ public boolean isPartitionSubscribed(String topicName, int partition) {
     return subscription.get();
   }
 
+  public void maybeSubscribeNewResource(String topicName, int partition) {
+    topicPartitionSubscriptionMap.putIfAbsent(topicName, new VeniceConcurrentHashMap<>());
+    topicPartitionSubscriptionMap.get(topicName).putIfAbsent(partition, new AtomicBoolean(true));
+  }
+
+  public boolean isNewResource(String topicName, int partition) {
+    AtomicBoolean subscription =
+        getTopicPartitionSubscriptionMap().getOrDefault(topicName, Collections.emptyMap()).get(partition);
+    return subscription == null;
```

Comment:
Since we are not going to use `subscription in any way, this would be a good use case for `containsKey`...

```
return getTopicPartitionSubscriptionMap().getOrDefault(topicName, Collections.emptyMap()).containsKey(partition);
```

Possibly put `containsKey` on the next line, if you'd like.

Reply:
Actually this method is not used after final fix. Just removed it.

Reply:
Even better :D